### ISSUE TITLE
InitSpace panic with const array size Option

### ISSUE THREAD

**Issue by donkeizluv** [coral-xyz/anchor#2556]
InitSpace account panics if there is an Option array field with const size

following snippet will panic on build:

```rust
const META_HASH_LEN: usize = 32;

#[account]
#[derive(InitSpace)]
pub struct Config {
    pub bump: u8,
    pub meta_hash: Option<[u8; META_HASH_LEN]>,
}
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }', lang/syn/src/idl/mod.rs:219:76
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Aursen** [comment]
Thanks for the issue, I will fix it

**Comment by acheroncrypto** [comment]
Fixed in #2555.
